### ISSUE TITLE
"Shift & Backspace" on input deletes selected node

### ISSUE THREAD

**Issue by eamador** [wbkd/react-flow#3416]
### Describe the Bug

When there is an input and you are typing holding _Shift_ (for instance. to capitalize text), and then you want to delete a part of the text, still holding _Shift_, the selected node will be deleted.

This was detected and can be reproduced in the react-flow examples and even react-flow's home page.

### Example App

https://codesandbox.io/s/awesome-roentgen-ndvw7w

### Steps to Reproduce the Bug or Issue

1. Go to https://codesandbox.io/s/awesome-roentgen-ndvw7w
2. Select / click any node
3. Click in the upper input in order to focus in it
4. Hold Shift + delete once
5. Hold Shift + delete again

https://github.com/wbkd/react-flow/assets/9006480/ca4abf9b-c73d-48ed-9d51-8214e989629f

### Expected behavior

As a user, I would like to be able to delete a text from an input, even if i am holding _Shift_ case, and don't get the selected node deleted.

### Screenshots or Videos

https://github.com/wbkd/react-flow/assets/9006480/b060226e-634e-4489-bafe-834981b66c54

https://github.com/wbkd/react-flow/assets/9006480/6f94f8d9-45b9-49bc-980b-e4f42d6eb4bc

### Platform

- OS: [macOS, Windows, Linux]
- Browser: [Chrome, Safari, Firefox]
- Version: [11.8.3]

### Additional context

It could get fixed by doing some changes in `packages/core/src/hooks/useKeyPress.ts` hook, specifically in the `downHandler`, which if changes the following, it fixes the problem.

```ts
      const downHandler = (event: KeyboardEvent) => {
        modifierPressed.current = event.ctrlKey || event.metaKey || event.shiftKey;

        if (!modifierPressed.current && isInputDOMNode(event)) {
          return false;
        }
        const keyOrCode = useKeyOrCode(event.code, keysToWatch);
        pressedKeys.current.add(event[keyOrCode]);

        const matchingKey = isMatchingKey(keyCodes, pressedKeys.current, false);
        if (matchingKey && !event.shiftKey) {
          event.preventDefault();
          setKeyPressed(true);
        }
      };
```

https://github.com/wbkd/react-flow/assets/9006480/67318419-f4f2-4b78-a6cd-79cc22a77f1e

**Comment by bcakmakoglu** [comment]
You can also fix this from your side by simply binding a keydown handler on the input field `onKeyDown={(e) => e.stopPropagation() && e.preventDefault()}`

[Here's an example.](https://codesandbox.io/s/jovial-chatterjee-tvxv42?file=/src/App.js:237-306)

**Comment by moklick** [comment]
thanks for the report @eamador - I will close this because it's a duplicate of #3328

**Comment by bcakmakoglu** [comment]
@eamador For sure, it's just a "patch" so that you can avoid this issue *until* it's actually fixed 😄

**Comment by moklick** [comment]
fixed in 11.9.0
